### ISSUE TITLE
add_missing_usings treats using-alias Name as imported namespace → SUCCESS miss + CS0246

### ISSUE THREAD

**Issue by JoshuaRamirez** [JoshuaRamirez/RoslynMcpServer#1349]
## Summary
`add_missing_usings` fails to add a needed namespace when the file already has a **using alias** whose right-hand `Name` equals that namespace (e.g. `using Foo = System.Collections.Generic;`). The operation finds the correct candidate (`System.Collections.Generic`) but then filters it out because it treats every `UsingDirectiveSyntax.Name` as an already-imported namespace — including aliases. Result: **SUCCESS with 0 changes** while `List<>` (etc.) remain **CS0246**.

## Master SHA probed
`ea6fe467cc9605acc57eb376ceaa4711829b08ac`

## Root cause
**File:** `src/RoslynMcp.Core/Refactoring/Organize/AddMissingUsingsOperation.cs`  
**Methods:** `ExecuteSingleFileAsync` (~L146–148) and `ExecuteAllFilesAsync` (~L270–272)

```csharp
var existingUsings = root.Usings.Select(u => u.Name?.ToString() ?? "").ToHashSet();
var newUsings = namespacesToAdd.Where(n => !existingUsings.Contains(n)).ToList();
```

For `using Foo = System.Collections.Generic;`, `u.Name` is `System.Collections.Generic` and `u.Alias` is `Foo`. An alias does **not** import the namespace into scope (only the alias identifier does), so this false-positive “already present” check skips the real `using System.Collections.Generic;` that would fix CS0246.

## Minimal repro

**Before:**
```csharp
using Foo = System.Collections.Generic;

class C
{
    void M()
    {
        List<int> xs = new List<int>();
    }
}
```

**Expected:** SUCCESS that adds `using System.Collections.Generic;` (alias alone does not bring `List<>` into scope).

**Actual:** SUCCESS with **0** usings added / no file change. Both `List<>` references stay **CS0246**.

Probe (same logic as the operation) against an AdhocWorkspace with BCL refs:
- Diagnostics: `CS0246` on both `List<>` sites
- `namespacesToAdd`: `System.Collections.Generic`
- After existing-`Name` filter: **empty**
- If aliases are excluded from `existingUsings`: correctly keeps `System.Collections.Generic`

## Compile errors
- `CS0246`: The type or namespace name `List<>` could not be found

## Why distinct from nearby open bugs
- **#1346** — `remove_unused_usings` incorrectly **strips** live using-static / aliases (opposite op; destroy vs miss-add).
- **#1347** — `remove_unused_usings` strips live **global** usings in `GlobalUsings.cs`.
- **#540** — `add_missing_usings` DocumentNotEditable (3115) mapping only; not SUCCESS/CS0246.
- **#1322** — `convert_foreach_linq` emits LINQ without adding `using System.Linq` (different op).

No prior open SUCCESS+CS0246 bug for `add_missing_usings` alias false-positive skip.
